src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -113,6 +106,12 @@ public async Task Start(CancellationToken cancellationToken)
                                     if (Logger.IsWarn) Logger.Warn($"Failure when executing request {t.Exception}");
                                 }
 
+                                if (cancellationToken.IsCancellationRequested)
+                                {
+                                    if (Logger.IsDebug) Logger.Debug("Ignoring sync response as shutdown is requested.");
+                                    return;
+                                }
```

Comment:
I would put this inside the below try to still have call to `Free(allocation)`. It probably won't matter for shutdown but potentially if we have other places where we would like to cancel the sync - yes.